### ISSUE TITLE
melonDS does not build on arm* Alpine Linux

### ISSUE THREAD

**Issue by nergzd723** [Arisotura/melonDS#982]
I'm trying to package melonDS into Alpine Linux repos for use with postmarketOS, mobile GNU/Linux operating system.

While building for x86 and x86_64 works fine, building for any arm arch(both armv7 and aarch64) breaks with some OpenGL errors.
Here is the error log:
[log.txt](https://github.com/Arisotura/melonDS/files/5914335/log.txt)

I'm not sure what this might be related to, maybe some mesa header issues? Or Qt OpenGL lib being broken? I tried building both with clang and gcc, didn't help.

**Comment by RSDuck** [comment]
could it be possible that regular OpenGL isn't available on that distro?

**Comment by nergzd723** [comment]
No, I'm pretty sure regular OpenGL works there. I can run mesa-demos just fine, glxgears works. It's weird that it only breaks on arm* arch.

**Comment by RSDuck** [comment]
yeah looks like it.

@WaluigiWare64 is working on OGL ES support though.

**Comment by WaluigiWare64** [comment]
I remember this happened with the flatpak build too.

Also don't expect the OpenGL ES support to be ready any time soon, it's a big hack currently.
https://github.com/Arisotura/melonDS/tree/opengl-es-renderer

**Comment by nergzd723** [comment]
Maybe I can build it without OpenGL support as a workaround? Or there isn't a way to do so?

**Comment by RSDuck** [comment]
unfortunately disabling the OpenGL support only disables it in the core, the frontend still uses desktop OpenGL is some places

**Comment by Quackdoc** [comment]
Does Arm have to be locked to GLES? Most Arm devices (hardware wise) support full OGL too under mesa (even raspberry pi supports OGL), GLES is a now rather niche requirement out side of android (Not saying GLES shouldn't be ported to, just that I dont see a necessity to lock out arm and OGL)

**Comment by nadiaholmquist** [comment]
No, for some reason some distros (including flatpak...) wants to build Qt targeting GLES on ARM devices, which breaks basically every Qt application that relies on OpenGL. Would be really nice if they didn't and just shipped gl4es for the devices that don't support non-ES OpenGL, but well... :(

**Comment by WaluigiWare64** [comment]
> Does Arm have to be locked to GLES? Most Arm devices (hardware wise) support full OGL too under mesa (even raspberry pi supports OGL), GLES is a now rather niche requirement out side of android (Not saying GLES shouldn't be ported to, just that I dont see a necessity to lock out arm and OGL)

The Raspberry Pi 4 only supports up till OpenGL 2.1, however the hardware is capable of GLES 3.2 (it is not exposed in the driver yet).

**Comment by nadiaholmquist** [comment]
> > Does Arm have to be locked to GLES? Most Arm devices (hardware wise) support full OGL too under mesa (even raspberry pi supports OGL), GLES is a now rather niche requirement out side of android (Not saying GLES shouldn't be ported to, just that I dont see a necessity to lock out arm and OGL)
> 
> The Raspberry Pi 4 only supports up til OpenGL 2.1, however the hardware is capable of GLES 3.2 (it is not exposed in the driver yet, the current driver supports up til GLES 3.1).

The hardware can most likely do desktop OpenGL 3.x and up too, for example the Mali T860 can with the open source driver, but it was never supported officially. melonDS runs fine on devices that don't support the needed GL version, you just obviously get limited to the software renderer.

**Comment by nergzd723** [comment]
> Does Arm have to be locked to GLES? Most Arm devices (hardware wise) support full OGL too under mesa (even raspberry pi supports OGL), GLES is a now rather niche requirement out side of android (Not saying GLES shouldn't be ported to, just that I dont see a necessity to lock out arm and OGL)
I contacted the maintainer of qt5-qtbase package in Alpine and got a reply saying that desktop OpenGL provides worse performance on mobile GPUs. I guess that's the reason of locking it to arm only.

**Comment by Quackdoc** [comment]
I guess the answer is either wait untill GLES, or change distros to one that that doesn't rely on a package that nonsensically restricts usage.

**Comment by WaluigiWare64** [comment]
On Ubuntu for example, they provide both desktop GL and GLES packages.
As an alternative, perhaps you could try using Qt  6 (once the Qt 6 support PR  #865 is merged).

**Comment by WaluigiWare64** [comment]
On Ubuntu for example, they provide both desktop GL and GLES packages.
As an alternative, perhaps you could try using Qt  6 (once the Qt 6 support PR  #865 is merged).
